src/language-js/embed.js

```diff
@@ -616,8 +612,16 @@ function printHtmlTemplateLiteral(
     }
   );
 
+  const shouldBreak = hasNewlineInRange(
```

Comment:
Can't approve this. This contradicts one of the main principles of Prettier, which states that the original formatting should be used as little as possible, only as the last resort. What about always using `softline`?

Reply:
It is place when we should respect original formatting, we don't know in that place and which tags will be inserted and the absence or presence of a space can potentially break the code

Reply:
For example:
```js
const template1 = html`<img src="avatar.png" alt="Avatar" style="width:200px">`;
const template2 = html`<img src="avatar.png" alt="Avatar" style="width:200px">`;

insertIntoDom(template1 + template2);
```

```js
const template1 = html`<img src="avatar.png" alt="Avatar" style="width:200px">`;
const template2 = html`
  <img src="avatar.png" alt="Avatar" style="width:200px">
`;

insertIntoDom(template1 + template2);
```

Output difference result - 
1. No spaces between images
2. Space between images

Reply:
"Presence of whitespace", which you're talking about, and "presence of a new line", which is checked in the proposed code, are two different things, aren't they?

Reply:
See?

**Prettier pr-7208**
[Playground link](https://deploy-preview-7208--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuc0DOMAE8C2AHAGwEN5MBeTACxhwIANMAeASxwHNM0AnMMgHRBEAbiSJcAdHihsA9AIB8mOgG4QAGhAQ8MZumSgxXCAHcACmIRpkgoRGYATdSABGXImADWcGAGU875mlkGC4AVzgNaloAdUpmeDR-MDgfS3jmIXiAT2swNCsNQLQ4LhhTNzYcImQAMyICYo0AKzQADwAhN09vHyIcOAAZQLha+saQFtafQLYCOABFUIh4UYaIkH8uYq5rZyJnOAInPC5AmGiHGEpkAA4ABg0TiGLotzxrE7htoRGNAEclvBylorCgiGgALRQOBweywpxcOAA5iI8pESrVJB1NYaYo4ZjBMLrNAzOaLZYjLFjdYwfYXexXZAAJg0ISIzAIMwAwhAcFVrFBoL8QKFigAVfag7HjIThACSUDhsB8YFO2gAgoqfDAsnNVsUAL4GoA)
```sh
--parser babel
```

**Input:**
```jsx
const template = html` <img src="avatar.png/"> `;
```

**Output:**
```jsx
const template = html`<img src="avatar.png/" />`;

```

Reply:
That is out of scope of this PR. If we want to keep that, we should turn off the whole HTML formatting.

Reply:
@saschanaz We're talking about keeping significant HTML whitespace or absence of it at the beginning and at the end of the template. It's okay to replace spaces and tabs with new lines and vice versa, but it's not okay to insert space/tab/new line where there was none. Whitespace inside the markup is already properly handled by the HTML formatter, we need only to take proper care of the edges of the template.

Reply:
Not sure what you mean. This PR prevents adding newline when not needed, and thus does not "insert space/tab/new line where there was none".

Reply:
@thorn0 `contentDoc` already contains code with added/removed whitespaces (`Presence of whitespace`), your example is wrong, just add more characters and prettier break code, changing `--html-whitespace-sensitivity` is not solve problem and should not do that 

https://deploy-preview-7208--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuc0DOMAE8C2AHAGwEN5MBeTACxhwIAMAeASxwHNM0AnMMgHRCIA3EkU4A6PFFb9MAegB8mXlDoBuEABoQEPDCbpkoUZwgB3AAqiEaZAMEQmAE00gARpyJgA1nBgBlPE8mKWQYTgBXOC1qWgB1SiZ4NECwOD9rRKZBRIBPWzA0Gy1gtDhOGHMPVhwiZAAzIgJSrQArNAAPACEPb18-Ihw4ABlguHrG5pA29r9g1gI4AEVwiHhxpqiQQM5SzltXIlc4Ahc8TmCYWKcYSmQADgAGLTOIUtiPPFszuF3BMa0AI4reCVHQ2FBENAAWigcDgjnhLk4cCBTGRlSI1VqSAaGy0pRwTFCEU2aDmC2WqzGOImmxghyujhuyAATFowkQmAQ5gBhCA4Gq2KDQf4gcKlAAqh3BuMmgkiAEkoAjYH4wOddABBZV+GA5BbrUoAXyNQA

https://deploy-preview-7208--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuc0DOMAE8C2AHAGwEN5MBeTACxhwIAMAeASxwHNM0AnMMgHRCIA3EkU4A6PFFb9MAegB8mAGYATNCqJot2nbr36ddANwgANCAh4YTdMlCjOEAO4AFUQjTIBgiExVmQACNOIjAAazgYAGU8UKYpZBhOAFc4c2paAHVKJng0WLA4KI9cpkFcgE8vMC0A+LQ4ThgXENYcImQlIgIG8wArNAAPACEQ8Mioohw4ABl4uE7u3pABwaj41gI4AEVkiHhFnrSQWM4Gzi9AokC4AgC8TniYTL8YSmQADgAGcweIBsyITwXgecHOggW5gAjnt4C1LJ4UJoALRQOBwFQYgKcOAwpg4lpENodJBdI7mBo4JiJFLHNAbLa7fYLUlLY4wa4vFRvZAAJnMSSITAIGwAwhAcO0vFBoJCQMkGgAVa6IsnLQSpACSUExsCiYEeVgAgjqojAKltDg0AL7WoA

Reply:
@evilebottnawi Let's take a look at your example:
```
const template1 = html`<img src="avatar.png" alt="Avatar" style="width:200px">`;
const template2 = html`<img src="avatar.png" alt="Avatar" style="width:200px">`;

insertIntoDom(template1 + template2);
```
With `--print-width 80` it's formatted like this (both in 1.19.1 and in this PR):
```
const template1 = html`
  <img src="avatar.png" alt="Avatar" style="width:200px" />
`;
const template2 = html`
  <img src="avatar.png" alt="Avatar" style="width:200px" />
`;

insertIntoDom(template1 + template2);
```
So now we get whitespace between the images in `insertIntoDom`. In the original code, there was no whitespace. Don't you think this is a problem?

Reply:
That's an existing behavior, right?

Out of curiosity, if you would format the following code:

```js
const template1 = html`<picture src="avatar.png" alt="Avatar" style="width:200px">This is an alternative content</picture>`;
```

would you prefer the following over the current behavior? (Both still insert a newline anyway):

```js
const template1 = html`<picture src="avatar.png" alt="Avatar" style="width:200px">
  This is an alternative content<
/picture>`;
```

Reply:
This is an existing behavior, and this PR unfortunately makes it worse (less deterministic) by making the result depend on `print-width`.

> would you prefer the following over the current behavior?

That would be a solution. We can think of other ways to format the output though, e.g.:
```js
const template1 = html
  `<picture src="avatar.png" alt="Avatar" style="width:200px"
    >This is an alternative content</picture
  >`;
```
Also I disagree with @evilebottnawi about `--html-whitespace-sensitivity` here. I think it should affect this. E.g. the behavior proposed in this PR would look logical with `--html-whitespace-sensitivity ignore`.

Reply:
@thorn0 yes, we should respect `html-whitespace-sensitivity` in this case

Reply:
Should I just add the check, or should we discuss more about the ways mentioned in https://github.com/prettier/prettier/pull/7208#discussion_r364711521? I personally lean to do better whitespace sensitive formatting.

Reply:
@thorn0 I think improving output as in your example of out scope this PR (we can open a new issue about it, i think we already have issue about it :smile: ), now we have bug and we need fix bug, broken code > improve readability, but yes we should respect `--html-whitespace-sensitivity`

Reply:
>I think improving output as in your example of out scope this PR

Improving as @thorn0 suggested will make this PR redundant, no? I would prefer better fix as I need a fix that works on whitespace sensitive mode.

Reply:
@saschanaz

> I personally lean to do better whitespace sensitive formatting.

That would be ideal. If you have the time to implement it, of course.

So we seem to have 4×2 cases:

1. no whitespace at both ends
2. whitespace only at the beginning
3. whitespace only at the end
4. whitespace at both ends

And every of them can fit or not fit on one line (×2).

Let's decide how these 8 cases should be formatted.

Reply:
@evilebottnawi This PR doesn't really fix that bug because the output depends on `print-width`. Add one more attribute to the markup or decrease `print-width`, and voila, the bug is back.

Reply:
@thorn0 I do not argue that this PR fix bug :confused:

Reply:
@evilebottnawi Sorry, for some reason I thought you were insisting on merging it as is.

Reply:
> no whitespace at both ends

I prefer my example in https://github.com/prettier/prettier/pull/7208#discussion_r364669394.

> whitespace only at the beginning
> whitespace only at the end
> whitespace at both ends

Just replace them with a single space character \u0020 on top of the above example?

> And every of them can fit or not fit on one line (×2).

I think those should just be kept, assuming there are not breakable.

Reply:
> I prefer my example in #7208 (comment).

The  weakness of that formatting is that the line can already contain a lot of stuff in front of the template literal. The name of the first tag gets too separated form the rest of the markup:

```js
const veryLongVariableNameVeryLongVariableNameVeryLongVariableName = html`<picture
  src="avatar.png"
  alt="Avatar"
  style="width:200px"
  >This is an alternative content</picture
>`;
```

Reply:
Anyway, as @evilebottnawi wrote, the main priority is to make the behavior correct. Readability can be improved later in separate PRs.

Reply:
Do you agree with my opinion in https://github.com/prettier/prettier/pull/7208#discussion_r364757079 except it can potentially be unreadable?

Reply:
As I wrote, the main priority is to make the behavior correct. This means I'd agree to any solution that is correct with regard to whitespace significance.

But if you really want to discuss the exact look and readability of those 10 cases, then for a substantive discussion we're going to need a code sample (at least one) for each of them as each of them has its own peculiarities. Also it's better to move this discussion back to #7096.

Reply:
Understood 👍

I'll just make the behavior correct and defer everything else.